nats-base-client/protocol.ts

```diff
@@ -217,6 +217,9 @@ export class ProtocolHandler implements Dispatcher<ParserEvent> {
     this.resetOutbound();
 
     const pong = deferred<void>();
+    pong.catch(() => {
+      // provide at least one catch - as pong rejection can happen before it is expected
```

Comment:
How about a log warning about such errors, or better make it properly awaited? There's no guarantee that it will always fail with an "ok" error. If some exception happens, it will be swallowed here, which will make debugging quite a challenge